consensus/src/quorum_store/network_listener.rs

```diff
@@ -85,12 +97,13 @@ impl NetworkListener {
                         request.source(),
                         request.digest()
                     );
-                    let cmd =
-                        BatchReaderCommand::GetBatchForPeer(request.digest(), request.source());
-                    self.batch_reader_tx
-                        .send(cmd)
-                        .await
-                        .expect("could not push Batch batch_reader");
+                    if let Ok(value) = self.batch_reader.get_batch_from_local(&request.digest()) {
+                        let batch =
+                            Batch::new(self.my_peer_id, self.epoch, request.digest(), value);
+                        self.network_sender
+                            .send_batch(batch, vec![request.source()])
+                            .await;
```

Comment:
Can the await be a performance problem since it delays handling messages on the critical path?